BookingProject/BookingWebAPI/Payment.Application/Services/PaymentService/PaymentService.cs

```diff
@@ -10,11 +11,28 @@
 
 namespace Payment.Application.Services.PaymentService
 {
-    public class PaymentService(IPaymentRepository _paymentRepository) : IPaymentService
+    public class PaymentService(
+        IPaymentRepository _paymentRepository,
+        IDistributedLockService _lockService
+    ) : IPaymentService
     {
         public async Task<Result<bool>> CompensateBalance(BalanceRequestDto balanceDto)
         {
-            return await _paymentRepository.CompensateBalance(balanceDto.UserId, balanceDto.Price, balanceDto.CurrencyName);
+            var lockKey = $"balance:{balanceDto.UserId}";
+            var lockValue = await _lockService.AcquireLockAsync(lockKey, TimeSpan.FromSeconds(10));
+            if (string.IsNullOrEmpty(lockValue))
+            {
+                return Result<bool>.Failure("Unable to acquire lock. Try again later.");
+            }
+
+            try
+            {
+                return await _paymentRepository.CompensateBalance(balanceDto.UserId, balanceDto.Price, balanceDto.CurrencyName);
+            }
+            finally
+            {
+                await _lockService.ReleaseLockAsync(lockKey, lockValue);
+            } 
```

Comment:
All this code should be in try finally, because if exception happens before entering try block, then the lock won't be released